src/index.js

```diff
@@ -0,0 +1,254 @@
+  }
+
+  function displayWinner() {
+    const announcementNode = document.querySelector('#announcement');
+    announcementNode.style.display = 'block';
+    announcementNode.textContent = `${this.currentPlayer.name} wins!`;
+  }
+
+  function displayDraw() {
+    const announcementNode = document.querySelector('#announcement');
+    announcementNode.style.display = 'block';
+    announcementNode.textContent = 'Draw!';
+  }
+
+  Game.prototype.run = function () {
+    const game = this;
+    game.bootstrap();
+
+    const cells = document.querySelectorAll('.cell');
+    cells.forEach((el) => {
+      el.addEventListener('click', (e) => {
+        const index = e.target.dataset.index - 1;
+        if (!game.board.get(index) && !game.finished) {
+          game.board.put(game.currentPlayer.symbol, index);
+          game.render();
+          const winning = new Evaluator(game.board, index + 1).isWinning();
+          if (winning) {
+            game.currentPlayer.score += 1;
+            game.displayPlayerInfo();
+            displayWinner.bind(game)();
+            game.finished = true;
+          } else if (game.board.isFilled()) {
+            displayDraw();
+          } else {
+            game.switchPlayers();
+          }
+        }
+      });
+    });
+
+    const settingBtn = document.querySelector('#settings');
+    settingBtn.addEventListener('click', () => {
+      const modal = document.querySelector('#modal');
+      modal.style.display = 'block';
+    });
+
+    const resetBtn = document.querySelector('#reset');
+    resetBtn.addEventListener('click', () => {
+      game.currentPlayer = game.playerX;
+      game.board = new Board();
+      game.finished = false;
+      game.render();
+      hideAnnouncement();
+    });
+  };
+
+  Game.prototype.bootstrap = function () {
+    const game = this;
+    document.getElementById('confirm').addEventListener('click', () => {
+      const p1 = document.querySelector('#p1input');
+      const p2 = document.querySelector('#p2input');
+
+      const p1Name = p1.value || p1.placeholder;
+      const p2Name = p2.value || p2.placeholder;
+
+      document.querySelector('#modal').style.display = 'none';
+      [game.playerX, game.playerO] = [new Player(p1Name, 'X'), new Player(p2Name, 'O')];
+      game.currentPlayer = game.playerX;
+      game.board = new Board();
+      game.finished = false;
+      game.render();
+      hideAnnouncement();
+
+      game.displayPlayerInfo();
+    });
+  };
+
+  Game.prototype.displayPlayerInfo = function () {
+    const p1NameNode = document.querySelector('#p1');
+    const p2NameNode = document.querySelector('#p2');
+    const p1ScoreNode = document.querySelector('#p1score');
+    const p2ScoreNode = document.querySelector('#p2score');
+
+    p1NameNode.textContent = this.playerX.name;
+    p1ScoreNode.textContent = this.playerX.score;
+
+    p2NameNode.textContent = this.playerO.name;
+    p2ScoreNode.textContent = this.playerO.score;
+  };
```

Comment:
Let's separate the DOM from application logic, you can create another file for DOM.

![image](https://user-images.githubusercontent.com/31897434/73420341-70e18b80-4322-11ea-840a-fa14e04b592b.png)

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/

Reply:
Resolved